04.bowling/bowling.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+score = ARGV[0]
+scores = score.to_s.split(',')
+shots = []
+scores.each do |s|
+  if s == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << s.to_i
+  end
+end
+
+frames = []
+shots.each_slice(2) do |s|
+  frames << s
+end
+indexes = [*(1..12)]
+i_f_hash = Hash[indexes.zip frames]
+
+n = 0
+point = 0
+i_f_hash.each do
```

Comment:
`each` を使っているのに、ループした時の要素を受け取らない理由はありますか？

```suggestion
i_f_hash.each do |index, frame|
```

という感じで受け取れるので、それらをうまく使えばコードをわかりやすく書けると思います。

Reply:
もしくは `i_f_hash` を使わなくても `frames` を使えばスコア計算ができそうです。
書きはサンプルです。全部書いてしまうと答えになってしまうので適度に省略しています。

```ruby
frames.each.with_index do |frame, index|
  break if index == 10
  next_frame = frames[index + 1]
  next_next_frame = frames[index + 2]

  if frame[0] == 10 # ストライク
    # 省略...
  elsif frame.sum == 10 # スペア
    # 省略...
  else
    point += frame.sum
  end
end

```